lldb-eval/parser.cc

```diff
@@ -2133,6 +2154,36 @@ ExprResult Parser::ParseBooleanLiteral() {
       /*is_literal_zero*/ false);
 }
 
+ExprResult Parser::ParseCharLiteral() {
+  ExpectOneOf(clang::tok::char_constant, clang::tok::wide_char_constant,
+              clang::tok::utf16_char_constant, clang::tok::utf32_char_constant);
+  clang::SourceLocation loc = token_.getLocation();
+
+  llvm::SmallString<8> CharBuffer;
+  bool CharInvalid = false;
+  llvm::StringRef ThisTok = pp_->getSpelling(token_, CharBuffer, &CharInvalid);
+  clang::CharLiteralParser charLiteral(ThisTok.begin(), ThisTok.end(), loc,
+                                       *pp_, token_.getKind());
+
+  if (CharInvalid || charLiteral.hadError()) {
+    BailOut(
+        ErrorCode::kInvalidCharLiteral,
+        "Failed to parse token as char-constant: " + TokenDescription(token_),
+        token_.getLocation());
+    return std::make_unique<ErrorNode>(ctx_->GetEmptyType());
+  }
+
+  auto basicType = PickCharType(charLiteral);
+  llvm::APInt literal_value(
+      ctx_->GetBasicType(basicType)->GetByteSize() * CHAR_BIT,
```

Comment:
`ctx_->GetBasicType(basicType)` is relatively cheap, but still it's better to resolve it only once -- save it into a variable.